src/app/src/advancedsetupdialog.cpp

```diff
@@ -0,0 +1,88 @@
+#include <format>
+#include <spdlog/spdlog.h>
+
+#include "advancedsetupdialog.h"
+#include "ui_advancedsetupdialog.h"
+
+/*
+* Instance of advanced setup options window.
+*
+* @param parent Pointer to parent widget.
+*/
+AdvancedSetupDialog::AdvancedSetupDialog( toml::value config,std::string* m_niDev,QWidget *parent) : QDialog(parent), ui(new Ui::AdvancedSetupDialog) {
+    ui->setupUi(this);
+
+    std::string niDev = toml::find_or<std::string>(config, "device", "nidaqmx","device", std::string("Dev2"));
+    new_niDev = niDev;
+    m_niDev=m_niDev;
```

Comment:
if you're passing in the config and reading the device name I don't think you need to also pass in the m_niDev, this class also shouldn't modify the `m_niDev` that is passed in, it should let `MainWindow` do that and just emit the signal with the new device name.